### PR TITLE
Fix the upgrade nudge

### PR DESCRIPTION
**Pull request by scruffian**
#### Changes proposed in this Pull Request
The upgrade nudge isn't displayed in devdocs:
- http://wpcalypso.wordpress.com/devdocs/blocks/upgrade-nudge
- http://wpcalypso.wordpress.com/devdocs/blocks/upgrade-nudge-expanded

#### Testing instructions
Checkout this branch and look in devdocs:
- http://calypso.localhost:3000/devdocs/blocks/upgrade-nudge
- http://calypso.localhost:3000/devdocs/blocks/upgrade-nudge-expanded

### CODE REVIEW

File: client/blocks/upgrade-nudge-expanded/docs/example.jsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+					subtitle={
+						"By upgrading to a Business Plan you'll enable advanced SEO features on your site."
+					}
+					highlightedFeature={ FEATURE_ADVANCED_SEO }
+					benefits={ [
+						"Preview your site's posts and pages as they will appear when shared on Facebook, Twitter and the WordPress.com Reader.",
+						'Allow you to control how page titles will appear on Google search results, or when shared on social networks.',
+						'Modify front page meta data in order to customize how your site appears to search engines.',
+					] }
+					shouldDisplay={ stubTrue }
```
Comment (jsnajdr) [inline client/blocks/upgrade-nudge-expanded/docs/example.jsx]: `shouldDisplay` is a boolean, but the `stubTrue` value is a function returning boolean.

File: client/blocks/upgrade-nudge/index.jsx [hunk 2]
```diff
@@ -100,7 +102,7 @@ export class UpgradeNudge extends React.Component {
 			( feature === FEATURE_NO_ADS && site.options.wordads ) ||
 			( ( ! jetpack && site.jetpack ) || ( jetpack && ! site.jetpack ) );
 
-		if ( shouldNotDisplay ) {
+		if ( shouldNotDisplay && ! shouldDisplay ) {
```
Comment (jsnajdr) [inline client/blocks/upgrade-nudge/index.jsx]: There's a refactoring opportunity based on the observation that most of `UpgradeNudge`'s props are used to compute `shouldNotDisplay` and nothing else. The condition could be therefore moved to the `connect` function:

Before:
```js
function UpgradeNudge( { a, b, c, d } ) {
  const shouldDisplay = complexCondition( a, b, c, d );
  ...
}

connect( state => ( {
  a: ...
  b: ...
  c: ...
  d: ...
} ) )( UpgradeNudge );
```

After:
```js
function UpgradeNudge( { shouldDisplay } ) {
  ...
}

connect( state => ( {
  shouldDisplay: complexCondition( a, b, c, d ),
} )( UpgradeNudge );
```

That makes the component more universal (has fewer props, `shouldDisplay` is usable in both devdocs and production).

**Comment by sixhours** [comment]
Works for me! `wp-desktop` test failure is unrelated. :shipit:

File: client/blocks/upgrade-nudge-expanded/index.jsx [hunk 3]
```diff
@@ -149,6 +149,7 @@ UpgradeNudgeExpanded.propTypes = {
 	eventName: PropTypes.string,
 	event: PropTypes.string,
 	siteSlug: PropTypes.string,
+	shouldDisplay: PropTypes.bool,
```
Comment (jsnajdr) [inline client/blocks/upgrade-nudge-expanded/index.jsx]: `forceDisplay` would be a better name. If `forceDisplay === false`, the component can still be displayed if `currentPlan` is provided. With `shouldDisplay === false`, the same sentence sounds a bit confusing.

**Comment by scruffian** [comment]
@jsnajdr thanks for the review. I've refactored based on your suggestion.

File: client/blocks/upgrade-nudge-expanded/docs/example.jsx [hunk 4]
```diff
@@ -0,0 +1,40 @@
+/** @format */
```
Comment (bluefuton) [inline client/blocks/upgrade-nudge-expanded/docs/example.jsx]: The format pragma is no longer needed and can be removed (see also https://github.com/Automattic/wp-calypso/pull/36353).

File: client/blocks/upgrade-nudge/docs/example.jsx [hunk 5]
```diff
@@ -17,14 +16,14 @@ const UpgradeNudgeExample = () => {
...
-				<UpgradeNudge feature={ FEATURE_CUSTOM_DOMAIN } href="#" shouldDisplay={ stubTrue } />
+				<UpgradeNudge feature={ FEATURE_CUSTOM_DOMAIN } href="#" forceDisplay={ true } />
 			</div>
 			<div>
 				<UpgradeNudge
 					title="This is a title"
 					message="This is a custom message"
 					icon="customize"
-					shouldDisplay={ stubTrue }
+					forceDisplay={ true }
```
Comment (jsnajdr) [inline client/blocks/upgrade-nudge/docs/example.jsx]: Nit: `<Upgrade forceDisplay />` is a nice shortcut for `true` props.

**Comment by scruffian** [comment]
@jsnajdr that refactor messes with the tests, which work by overriding the props. If there's a way round that then I'm happy to make it, but I'll do it in a different PR.

**Pull request merged**